src/inspect.mjs

```diff
@@ -134,71 +583,119 @@ const INSPECTORS = {
+        }
       }
 
-      const keys = X(v.OwnPropertyKeys());
       const cache = [];
       for (const key of keys) {
         const C = X(v.GetOwnProperty(key));
-        if (C.Enumerable === Value.true) {
-          cache.push([
-            Type(key) === 'String' && bareKeyRe.test(key.stringValue()) ? key.stringValue() : i(key),
-            C.Value ? i(C.Value) : '<accessor>',
-          ]);
-        }
+        const k = Type(key) === 'String' && bareKeyRe.test(key.stringValue())
+          ? ctx.stylize(key.stringValue(), 'name')
+          : i(key);
+        cache.push([
+          C.Enumerable === Value.true
+            ? k
+            : `${ctx.stylize(`[${k}`, 'hidden')}${ctx.stylize(']', 'hidden')}`,
+          formatPropertyDescriptor(C, ctx, i),
+        ]);
       }
 
-      const tag = getObjectTag(v);
-      let out = tag && tag !== 'Object' ? `${tag} {` : '{';
-      if (cache.length > 5) {
+      let outStr = `${base ? `${base} ` : ''}${braces[0]}`;
+      if (slots.length > 0 || (cache.length + outArray.length) > 5) {
+        slots.forEach((s) => {
+          outStr += `\n${'  '.repeat(ctx.indent)}${ctx.stylize(s[0], 'internalSlot')}: ${s[1]}`;
+        });
+        outArray.forEach((c) => {
+          outStr += `\n${'  '.repeat(ctx.indent)}${c},`;
+        });
         cache.forEach((c) => {
-          out = `${out}\n${'  '.repeat(ctx.indent)}${c[0]}: ${c[1]},`;
+          outStr += `\n${'  '.repeat(ctx.indent)}${c[0]}: ${c[1]},`;
         });
-        return `${out}\n${'  '.repeat(ctx.indent - 1)}}`;
+        outStr += `\n${'  '.repeat(ctx.indent - 1)}${braces[1]}`;
       } else {
         const oc = ctx.compact;
         ctx.compact = true;
+        outArray.forEach((c, index) => {
+          outStr = `${outStr}${(index === 0) ? '' : ','} ${c}`;
+        });
         cache.forEach((c, index) => {
-          out = `${out}${index === 0 ? '' : ','} ${c[0]}: ${c[1]}`;
+          outStr = `${outStr}${(index === 0 && outArray.length === 0) ? '' : ','} ${c[0]}: ${c[1]}`;
         });
         ctx.compact = oc;
-        return `${out} }`;
+        outStr += ` ${braces[1]}`;
       }
+
+      return outStr;
     } catch {
-      return compactObject(ctx, v);
+      return compactObject(v, constructor, tag);
     } finally {
       ctx.indent -= 1;
       ctx.inspected.pop();
     }
   },
 };
 
-export function inspect(value) {
+const hasOwn = Function.prototype.call.bind(Object.prototype.hasOwnProperty);
+const getOption = (options, name, convert) => {
+  Assert(typeof name === 'string' && name in defaultOptions);
+  Assert(typeof convert === 'function');
+
+  if (hasOwn(options, name)) {
+    return convert(options[name]);
+  } else {
+    return defaultOptions[name];
+  }
+};
+
+export function inspect(value, options = {}) {
   const context = {
     realm: surroundingAgent.currentRealmRecord,
     indent: 0,
     inspected: [],
+    stylize: getOption(options, 'colors', Boolean) ? stylizeWithColor : stylizeNoColor,
```

Comment:
you can replace these `getOption` things with just `options.colors ? stylizeWithColors : stylizeNoColor`, we don't need to worry about adversarial input like node does.

Reply:
Well, `getOption` falls back to the `defaultOptions` object.

Reply:
they're all false, and not passing `options.whatever` is falsy, so it should work.